### PR TITLE
chore(sentry-sdk): add traces

### PR DESCRIPTION
**Pull request by guidoiaquinti**
## Problem
While working on #10743 I've realised we are running without traces enabled. 

## Changes
Set the `traces_sample_rate` to 0.1% (we will bump the latter little by little after monitoring the usage).

## How did you test this code?
I didn't.

### CODE REVIEW

File: posthog/settings/sentry.py [hunk 1]
```diff
@@ -16,8 +17,16 @@
         sentry_logging = sentry_logging = LoggingIntegration(level=logging.INFO, event_level=None)
         sentry_sdk.init(
             dsn=os.environ["SENTRY_DSN"],
-            integrations=[DjangoIntegration(), CeleryIntegration(), RedisIntegration(), sentry_logging],
-            request_bodies="always",
-            send_default_pii=True,
             environment=os.getenv("SENTRY_ENVIRONMENT", "production"),
+            sample_rate=1.0,  # Configures the sample rate for error events, in the range of 0.0 to 1.0. The default is 1.0 which means that 100% of error events are sent. If set to 0.1 only 10% of error events will be sent. Events are picked randomly.
```
Comment (guidoiaquinti) [inline posthog/settings/sentry.py:21]: The SDK default was already 1.0 (see code comment)

File: posthog/settings/sentry.py [hunk 2]
```diff
@@ -16,8 +16,10 @@
         sentry_logging = sentry_logging = LoggingIntegration(level=logging.INFO, event_level=None)
         sentry_sdk.init(
             dsn=os.environ["SENTRY_DSN"],
+            environment=os.getenv("SENTRY_ENVIRONMENT", "production"),
             integrations=[DjangoIntegration(), CeleryIntegration(), RedisIntegration(), sentry_logging],
             request_bodies="always",
+            sample_rate=1.0,  # Configures the sample rate for error events, in the range of 0.0 to 1.0. The default is 1.0 which means that 100% of error events are sent. If set to 0.1 only 10% of error events will be sent. Events are picked randomly.
             send_default_pii=True,
-            environment=os.getenv("SENTRY_ENVIRONMENT", "production"),
+            traces_sample_rate=0.1,  # A number between 0 and 1, controlling the percentage chance a given transaction will be sent to Sentry. (0 represents 0% while 1 represents 100%.) Applies equally to all transactions created in the app.
```
Comment (macobo) [inline posthog/settings/sentry.py:24]: This is 10% not 0.1% as advertised in the PR.

**Comment by macobo** [comment]
I'm lacking context why we care about traces here? :) What is the problem we're solving?

File: posthog/settings/sentry.py (hunk 2 above)
Comment (guidoiaquinti) [reply posthog/settings/sentry.py:24]: True brainfart 😄 thanks for catching this

**Comment by guidoiaquinti** [comment]
> I'm lacking context why we care about traces here? :) What is the problem we're solving?

Yesterday @timgl was trying to investigate a performance issue reported by a customer. As we unfortunately do not have a tracing solution available (yet) the task became more difficult than expected.

Mid-long term plan is to have a tracing solution available via our Helm chart but in the meantime let's leverage [what Sentry can provide](https://docs.sentry.io/product/sentry-basics/tracing/distributed-tracing/) us.

File: posthog/settings/sentry.py [hunk 3]
```diff
@@ -16,8 +16,10 @@
         sentry_logging = sentry_logging = LoggingIntegration(level=logging.INFO, event_level=None)
         sentry_sdk.init(
             dsn=os.environ["SENTRY_DSN"],
+            environment=os.getenv("SENTRY_ENVIRONMENT", "production"),
             integrations=[DjangoIntegration(), CeleryIntegration(), RedisIntegration(), sentry_logging],
             request_bodies="always",
+            sample_rate=1.0,  # Configures the sample rate for error events, in the range of 0.0 to 1.0. The default is 1.0 which means that 100% of error events are sent. If set to 0.1 only 10% of error events will be sent. Events are picked randomly.
             send_default_pii=True,
-            environment=os.getenv("SENTRY_ENVIRONMENT", "production"),
+            traces_sample_rate=0.01,  # A number between 0 and 1, controlling the percentage chance a given transaction will be sent to Sentry. (0 represents 0% while 1 represents 100%.) Applies equally to all transactions created in the app.
```
Comment (macobo) [inline posthog/settings/sentry.py:24]: That's 1% - PR description mentioned 0.1% which would be 0.001
```suggestion
            traces_sample_rate=0.001,  # A number between 0 and 1, controlling the percentage chance a given transaction will be sent to Sentry. (0 represents 0% while 1 represents 100%.) Applies equally to all transactions created in the app.
```

Comment (guidoiaquinti) [reply posthog/settings/sentry.py:24]: True brainfart #2 🤣 I thought I opened the PR yesterday mentioning 1%. I clearly need a coffee this morning.

**Review by macobo — APPROVED** [review]
I don't think this will be of any use debugging performance issues we typically run into but also not going to block this as you think it does :)

Approved pending fixing the percentage.

**Pull request merged**